library/ssl_misc.h

```diff
@@ -1922,8 +1922,12 @@ static inline const void *mbedtls_ssl_get_sig_algs(
 #if defined(MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED)
 
 #if !defined(MBEDTLS_DEPRECATED_REMOVED)
-    if( ssl->handshake != NULL && ssl->handshake->sig_algs != NULL )
+    if( ssl->handshake->sig_algs_heap_allocated == 1 &&
```

Comment:
Why do you add this check ? Otherwise, if we keep it,  it should be after the check `ssl->handshake != NULL`.

Reply:
https://github.com/Mbed-TLS/mbedtls/blob/e87aaf5ecae7f52a1928e30cde2dd808cc6c7c0b/library/ssl_tls.c#L856 will set `ssl->handshake->sig_algs`,  and `ssl->handshake->sig_algs_heap_allocated` is not set

Reply:
Yes but it is set to ` ssl->conf->sig_algs` which is the default return value of `mbedtls_ssl_get_sig_algs` thus not really needed I'd say.

Reply:
Yes. It is not needed